### ISSUE TITLE
 Clustermesh cluster connection is abnormal     clustermesh  can't  attach  two  clusters  

### ISSUE THREAD

**Issue by 15927561940** [cilium/cilium#33265]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After creating two Kubernetes clusters under the same VPC on Alibaba Cloud, I attempted to use Cilium Clustermesh to connect the pod networks of the two nodes. However, I discovered that while I can resolve the IPs generated by MetalLB, they are unreachable via telnet. The Cilium Clustermesh cluster connection is abnormal, despite having confirmed that all security group rules are set to allow traffic.

### Cilium Version

root@node02:~/cilium# cilium  version
cilium-cli: v0.16.10 compiled with go1.22.4 on linux/amd64
cilium image (default): v1.15.5
cilium image (stable): v1.15.6
cilium image (running): 1.15.6

root@node02:~/cilium# helm ls -A
NAME   	NAMESPACE     	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
cilium 	kube-system   	4       	2024-06-20 06:48:06.92243168 +0800 CST 	deployed	cilium-1.15.6 	1.15.6
metallb	metallb-system	1       	2024-06-20 06:19:48.793677708 +0800 CST	deployed	metallb-0.14.5	v0.14.5

### Kernel Version

root@node02:~/cilium# uname -ra
Linux node02 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

ubuntu2204

### Kubernetes Version

root@node02:~/cilium# kubectl  get nodes
NAME     STATUS   ROLES    AGE   VERSION
node02   Ready    master   8h    v1.28.1

also  the node

### Relevant log output

```shell
root@node02:~/cilium# ^C
root@node02:~/cilium# cilium   clustermesh  connect  --context  context-cluster2  --destination-context  context-cluster3
✨ Extracting access information of cluster cluster3...
🔑 Extracting secrets from cluster cluster3...
ℹ️  Found ClusterMesh service IPs: [172.25.80.200]
✨ Extracting access information of cluster cluster2...
🔑 Extracting secrets from cluster cluster2...
ℹ️  Found ClusterMesh service IPs: [172.25.80.100]
ℹ️ Configuring Cilium in cluster 'cluster2' to connect to cluster 'cluster3'
ℹ️ Configuring Cilium in cluster 'cluster3' to connect to cluster 'cluster2'
✅ Connected cluster cluster2 and cluster3!


root@node02:~/cilium# cilium   clustermesh  status
✅ Service "clustermesh-apiserver" of type "LoadBalancer" found
✅ Cluster access information is available:
  - 172.25.80.100:2379
✅ Deployment clustermesh-apiserver is ready
ℹ️  KVStoreMesh is disabled

⚠️  1/1 nodes are not connected to all clusters [min:0 / avg:0.0 / max:0]

🔌 Cluster Connections:
  - cluster3: 1/1 configured, 0/1 connected

🔀 Global services: [ min:0 / avg:0.0 / max:0 ]

❌ 1 Errors:
  ❌ cilium-mkk5z is not connected to cluster cluster3: Waiting for initial connection to be established
     💡 Run 'kubectl exec -it -n kube-system cilium-mkk5z -c cilium-agent -- cilium-dbg troubleshoot clustermesh cluster3' to investigate the cause


root@node02:~/cilium# cilium   clustermesh  status
✅ Service "clustermesh-apiserver" of type "LoadBalancer" found
✅ Cluster access information is available:
  - 172.25.80.100:2379
✅ Deployment clustermesh-apiserver is ready
ℹ️  KVStoreMesh is disabled

⚠️  1/1 nodes are not connected to all clusters [min:0 / avg:0.0 / max:0]

🔌 Cluster Connections:
  - cluster3: 1/1 configured, 0/1 connected

🔀 Global services: [ min:0 / avg:0.0 / max:0 ]

❌ 1 Errors:
  ❌ cilium-mkk5z is not connected to cluster cluster3: Waiting for initial connection to be established
     💡 Run 'kubectl exec -it -n kube-system cilium-mkk5z -c cilium-agent -- cilium-dbg troubleshoot clustermesh cluster3' to investigate the cause
root@node02:~/cilium# kubectl exec -it -n kube-system cilium-mkk5z -c cilium-agent -- cilium-dbg troubleshoot clustermesh cluster3
Found 1 remote cluster configurations
Troubleshooting filtered subset of clusters: cluster3

Remote cluster "cluster3":
📄 Configuration path: /var/lib/cilium/clustermesh/cluster3

🔌 Endpoints:
   - https://cluster3.mesh.cilium.io:2379
     ✅ Hostname resolved to: 172.25.80.200
     ❌ Cannot establish TCP connection to cluster3.mesh.cilium.io:2379: dial tcp 172.25.80.200:2379: i/o timeout

🔑 Digital certificates:
   ✅ TLS Root CA certificates:
      - Serial number:       52:7e:f0:59:44:20:e9:fb:83:03:02:59:6f:88:a7:4d
        Subject:             CN=Cilium CA
        Issuer:              CN=Cilium CA
        Validity:
          Not before:  2024-06-19 22:16:22 +0000 UTC
          Not after:   2027-06-19 22:16:22 +0000 UTC
   ✅ TLS client certificates:
      - Serial number:       60:32:72:63:10:46:c7:81:c8:1d:fa:c5:c3:7b:12:5b:f1:23:1d:f0
        Subject:             CN=remote
        Issuer:              CN=Cilium CA
        Validity:
          Not before:  2024-06-19 22:43:00 +0000 UTC
          Not after:   2027-06-19 22:43:00 +0000 UTC

⚙️ Etcd client:
   ❌ Failed to establish connection: context deadline exceeded

root@node02:~/cilium# kubectl -n kube-system exec -ti cilium-mkk5z -- cilium-health status
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), clean-cilium-state (init), install-cni-binaries (init)
Probe time:   2024-06-19T23:06:09Z
Nodes:
  cluster2/node02 (localhost):
    Host connectivity to 172.25.80.2:
      ICMP to stack:   OK, RTT=231.813µs
      HTTP to agent:   OK, RTT=291.149µs
    Endpoint connectivity to 172.0.0.123:
      ICMP to stack:   OK, RTT=267.205µs
      HTTP to agent:   OK, RTT=329.638µs







root@node02:~/cilium# kubectl -n kube-system exec -ti cilium-mkk5z -- bash
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), clean-cilium-state (init), install-cni-binaries (init)
root@node02:/home/cilium# cilium-dbg monitor --type drop
Listening for events on 8 CPUs with 64x4096 of shared memory
Press Ctrl-C to quit
time="2024-06-19T23:07:55Z" level=info msg="Initializing dissection cache..." subsys=monitor
xx drop (Unsupported L3 protocol) flow 0x0 to endpoint 0, ifindex 27, file bpf_lxc.c:1496, , identity 177135->unknown: fe80::b494:61ff:fe15:f42 -> ff02::2 RouterSolicitation
```

### Anything else?

I  have  tested   over   5  times

k8s  node02    mteallb   172.25.80.100:2379

k8s  node03    mteallb   172.25.80.200:2379

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 15927561940** [comment]
k8s02  podcidr   # K8S Service CIDR, not overlap with node(host) networking
SERVICE_CIDR="10.62.0.0/16"

# Cluster CIDR (Pod CIDR), not overlap with node(host) networking
CLUSTER_CIDR="172.22.0.0/16"

# NodePort Range
NODE_PORT_RANGE="30000-32767"




k8s03  podcidr# K8S Service CIDR, not overlap with node(host) networking
SERVICE_CIDR="10.63.0.0/16"

# Cluster CIDR (Pod CIDR), not overlap with node(host) networking
CLUSTER_CIDR="172.23.0.0/16"

# NodePort Range
NODE_PORT_RANGE="30000-32767"

# Cluster DNS Domain
CLUSTER_DNS_DOMAIN="cluster.local"



cilium  cidr  with  helm---k8snode02
ipam:
  # -- Configure IP Address Management mode.
  # ref: https://docs.cilium.io/en/stable/network/concepts/ipam/
  mode: "cluster-pool"
  # -- Maximum rate at which the CiliumNode custom resource is updated.
  ciliumNodeUpdateRate: "15s"
  operator:
    # -- IPv4 CIDR list range to delegate to individual nodes for IPAM.
    clusterPoolIPv4PodCIDRList: ["172.0.0.0/8"]

ipv4NativeRoutingCIDR: "172.0.0.0/8"



k8snode03  is  the same



by the  way,ihave  tested  differernt  podcidr  3times

**Comment by giorio94** [comment]
Hi @15927561940,

The error is that the node in the first cluster cannot access the clustermesh-apiserver in the second one:

```
- https://cluster3.mesh.cilium.io:2379
     ✅ Hostname resolved to: 172.25.80.200
     ❌ Cannot establish TCP connection to cluster3.mesh.cilium.io:2379: dial tcp 172.25.80.200:2379: i/o timeout
```

This typically indicates issues in the underlying VPC or security groups configuration, or in the LoadBalancer settings. I'd suggest trying to ssh into the node and verifying that you can correctly curl `172.25.80.200:2379`, as well as that the clustermesh-apiserver pod is correctly up and running in both clusters.
